server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/metric/MetricImpl.java

```diff
@@ -41,13 +41,14 @@
   private final Integer decimalScale;
   private final Double bestValue;
   private final boolean bestValueOptimized;
+  private final boolean historyStored;
 
-  public MetricImpl(int id, String key, String name, MetricType type) {
-    this(id, key, name, type, null, null, false);
+  public MetricImpl(int id, String key, String name, MetricType type, boolean historyStored) {
+    this(id, key, name, type, null, null, false, historyStored);
   }
 
   public MetricImpl(int id, String key, String name, MetricType type, @Nullable Integer decimalScale,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)